### PR TITLE
systemd: lock down privileges more

### PR DESCRIPTION
**Pull request by batrick**
Adding a few more directives to lock down privileges more in systemd containers.

### CODE REVIEW

**Comment by batrick** [comment]
See `systemd.exec(5)` for information on these directives.

**Comment by batrick** [comment]
Tested this in Linode without issue.

File: systemd/ceph-osd@.service.in [hunk 1]
```diff
@@ -12,7 +12,14 @@ Environment=CLUSTER=ceph
 ExecStart=/usr/bin/ceph-osd -f --cluster ${CLUSTER} --id %i --setuser ceph --setgroup ceph
 ExecStartPre=/usr/lib/ceph/ceph-osd-prestart.sh --cluster ${CLUSTER} --id %i
 ExecReload=/bin/kill -HUP $MAINPID
+LockPersonality=true
+MemoryDenyWriteExecute=true
+NoNewPrivileges=true
```
Comment (liewegas) [inline systemd/ceph-osd@.service.in]: I think we need this one.  the osd and mon sudo smartctl (see ceph.git/sudoers.d)

File: systemd/ceph-mon@.service.in [hunk 2]
```diff
@@ -17,8 +17,14 @@ EnvironmentFile=-@SYSTEMD_ENV_FILE@
 Environment=CLUSTER=ceph
 ExecStart=/usr/bin/ceph-mon -f --cluster ${CLUSTER} --id %i --setuser ceph --setgroup ceph
 ExecReload=/bin/kill -HUP $MAINPID
+LockPersonality=true
+MemoryDenyWriteExecute=true
+NoNewPrivileges=true
```
Comment (liewegas) [inline systemd/ceph-mon@.service.in]: need this for sudo smartctl

**Comment by liewegas** [comment]
I think the rest are okay!

File: systemd/ceph-osd@.service.in (hunk 1 above)
Comment (batrick) [reply systemd/ceph-osd@.service.in]: done

**Comment by batrick** [comment]
Marking this needs-qa but I don't think any of our upstream testing is exercising the systemd units. I already did some basic tests on linode.

**Pull request merged**